File: Algebra_DivideFraction.cpp
// Algebra.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "DivideFraction.h"

#include <iostream>
#include <vector>
#include <exception>

using namespace std;

int main()
{
	cout << "\n    Welcome to our demonstration sequels";
	cout << "\n    Hope you enjoy (and follow) the lessons.\n\n";

	vector<unsigned> numerators;
	vector<unsigned> denominators;

	try {
		/*
		* Dividing fractions
		*/
		numerators = { 16, 9, 640, 7 };
		denominators = { 9, 20, 27, 20 };

		cout << "\n    Solving:\n";
		// Print as fraction
		for (unsigned n : numerators) {
			printf("%13u", n);
		}
		printf("\n%12s", " ");
		for (unsigned i = 0; i < numerators.size() - 1; i++) {
			cout << "-     /      ";
		}
		printf("%2s\n", "-");
		for (unsigned d : denominators) {
			printf("%13u", d);
		}
		cout << "\n";

		// use the DivideFraction class
		DivideFraction div_fract(numerators, denominators);
		div_fract.doDivide();

		printf("\n%25u\n", div_fract.answer[0]);
		printf("%25s\n", "Answer =    -");
		printf("%25u\n", div_fract.answer[1]);

		cout << "\n\n";

	} catch (exception& e) {
		cout << "\n" << e.what() << "\n";
	}

    return 0;
}

File: Algebra_DivideFraction.vb
Module Algebra_divide_fraction

    Sub Main()
        ''
        ' Dividing fractions
        ''
        Dim numerators = {16, 9, 640, 7}
        Dim denominators = {9, 20, 27, 20}
        Dim fractions As New Dictionary(Of String, Integer())
        fractions.Add("numerators", numerators)
        fractions.Add("denominators", denominators)

        Console.WriteLine("    Solving:")
        ' Print as fraction
        For Each numerator In fractions.Item("numerators")
            Console.Write(String.Format("{0,13}", numerator))
        Next
        Console.Write(Environment.NewLine & String.Format("{0,12}", " "))
        For wasted = 0 To numerators.Count - 2
            Console.Write(String.Format("{0}", "-     /      "))
        Next
        Console.WriteLine(String.Format("{0,1}", "-"))
        For Each denominator In fractions.Item("denominators")
            Console.Write(String.Format("{0,13}", denominator))
        Next
        Console.WriteLine("")

        ' use the DivideFraction Class
        Dim div_fract As New DivideFraction
        div_fract._init_(fractions)
        Dim fraction = div_fract.doDivide()

        Console.WriteLine(Environment.NewLine)

        Console.WriteLine(String.Format("{0,25}", fraction.Item("numerator")))
        Console.WriteLine(String.Format("{0,25}", "Answer   =   -"))
        Console.WriteLine(String.Format("{0,25}", fraction.Item("denominator")))

    End Sub

End Module
